### ISSUE TITLE
Using i18n-js causes Cannot read property 'prototype' of undefined error

### ISSUE THREAD

**Issue by PPHelios** [expo/expo#27060]
### Minimal reproducible example

https://snack.expo.dev/P7zlTmKurhazG7mvlNsX5

### Summary

I'm trying to use expo with i18n-js , I faced this issue in "expo-template-tabs" so i tried  to run the localization example on the expo docs page in snack on expo go directly from the website and also downloaded the repo, in both cases it gives same error.

I attached imgs of the errors on expo go when running from snack or npx expo start.

also find bellow the error logged in console.

` TypeError: Cannot read property 'prototype' of undefined, js engine: hermes
 ERROR  Invariant Violation: "main" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called., js engine: hermes
Android Bundling complete 78ms
 ERROR  TypeError: Cannot read property 'prototype' of undefined, js engine: hermes
 ERROR  Invariant Violation: "main" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called., js engine: hermes
`

![423062894_899957651669834_5056120888786668126_n](https://github.com/expo/expo/assets/103271591/4da8b6d3-bf82-4e70-8b80-ab437c5f32da)
![423541849_1417510118862255_3606403384306652216_n](https://github.com/expo/expo/assets/103271591/37d4c58f-2780-479a-ac00-2a91bca894d6)
![423221174_2359597490890894_2020615244582255307_n](https://github.com/expo/expo/assets/103271591/280bf892-4c08-4588-a109-1e566d323cac)

### Environment

 expo-env-info 1.2.0 environment info:
    System:
      OS: Windows 11 10.0.22631
    Binaries:
      Node: 20.10.0 - C:\Program Files\nodejs\node.EXE
      npm: 10.2.3 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      expo: ~49.0.18 => 49.0.22
      react: 18.2.0 => 18.2.0
      react-native: 0.72.6 => 0.72.6
    Expo Workflow: managed

**Comment by fayomihorace** [comment]
Yes, facing the same issue since yesterday. We have been using `i18n-js` in our expo project for 3 months now, and it has been working well. And just yesterday it stopped working.

**Comment by MrCox007** [comment]
Had to force version to  "i18n-js": "4.2.2"

**Comment by brentvatne** [comment]
it sounds like i18n-js may have done a new release that depends on a web api that doesn't exist in the react-native runtime. a good workaround for now is to use a previous version, like @MrCox007 suggested in https://github.com/expo/expo/issues/27060#issuecomment-1941429692. you may want to also follow up with i18n-js and create an issue on their repository to let them know about this regression in their library.

**Comment by Moonko** [comment]
I'm having the same issue with @tanstack/react-query and "@amplitude/analytics-react-native after updating to Expo 50 & react native 0.73.4

**Comment by BrazilianJoe** [comment]
Hello there! 

I digged a bit into the code.

When starting up a fresh project using: 
`npx create-expo-stack@latest poster --tamagui --expo-router --supabase --i18next`

On `languageDetector.ts`, these lines returns an error when testing or trying to build for production `Expo web`: 


```
const locales = Localization.getLocales();
const firstLanguageCode = locales[0].languageCode ?? 'en';
```

`Localization.getLocales()` returns an empty array `[]`. This is not in accordance to the documentation. 
As a result, the next line causes an error when testing Expo web.


https://docs.expo.dev/versions/latest/sdk/localization/#localizationgetlocales
_List of user's locales, returned as an array of objects of type Locale. Guaranteed to contain at least 1 element._

![Image](https://github.com/user-attachments/assets/07d7d1e2-a63c-4bd1-8295-0ee88a6a3a5a)

Digging deeper:

In the module `expo-localization/ExpoLocalization.ts`, getLocales start with: 
```
  getLocales(): Locale[] {
    const locales = getNavigatorLocales();
```

`getNavigatorLocales` is defined on the same file:
```
const getNavigatorLocales = () => {
  return Platform.isDOMAvailable ? navigator.languages || [navigator.language] : [];
};
```

On `expo-modules-core/src/Platform.ts`, `Platform` defines the `isDOMAvailable` property:

 ```
 isDOMAvailable,
  /**
   * Denotes if the current environment can attach event listeners
   * to the window. This will return false in native React
   * runtimes and Node.js.
   */
```

This is causing the inconsistent behavior on `web`, which is not in acordance to the documentation.

**Comment by aleqsio** [comment]
Hi @BrazilianJoe, thanks for the investigation!

On web platforms it should read these values from `packages/expo-modules-core/src/environment/browser.web.ts` – maybe this is caused by some invalid metro setup?
